### PR TITLE
added support to the accuracy metric for binary classification

### LINKED ISSUE #4456: Accuracy doesn't work for simple binary classifier
Hello,

This is quite a simple issue but the function `nnx.metric.Accuracy`  only works for a network that returns more than one output, i.e. 
`super().update(values=(logits.argmax(axis=-1) == labels))`. Not sure if this is by design. 

I was playing around with a simple binary classifier than only returned a single logit per observation. The Accuracy function gives nonsense for this case.

Anyway, might be worth noting in the doc string what the function expects.

### PR DESCRIPTION
**Pull request by mattbahr**
Fixes #4456 

Adds support for binary classification to the accuracy metric.

## Checklist
- [x] This PR fixes a minor issue (e.g.: typo or small bug) or improves the docs (you can dismiss the other
      checks if that's the case).
- [x] This change is discussed in a Github issue/
      [discussion](https://github.com/google/flax/discussions) (please add a
      link).
- [x] The documentation and docstrings adhere to the
      [documentation guidelines](https://github.com/google/flax/blob/main/docs/README.md#how-to-write-code-documentation).
- [x] This change includes necessary high-coverage tests.
      (No quality testing = no merge!)

### CODE REVIEW

File: flax/nnx/training/metrics.py [hunk 1]
```diff
@@ -240,26 +240,40 @@ class Accuracy(Average):
...
+
+    >>> logits3 = jax.random.normal(jax.random.key(2), (5,))
+    >>> labels3 = jnp.array([0, 1, 0, 1, 1])
+    >>> metrics.update(logits=logits3, labels=labels3)
+    >>> metrics.compute()
+    Array(0.8, dtype=float32)
   """
 
-  def update(self, *, logits: jax.Array, labels: jax.Array, **_) -> None:  # type: ignore[override]
+  def update(self, *, logits: jax.Array, labels: jax.Array, threshold=0.5, **_) -> None:  # type: ignore[override]
```
Comment (cgarciae) [inline flax/nnx/training/metrics.py:251]: This parameter should be a constructor argument instead.

File: flax/nnx/training/metrics.py [hunk 2]
```diff
@@ -240,26 +240,40 @@ class Accuracy(Average):
...
-        f'Expected logits.ndim==labels.ndim+1, got {logits.ndim} and {labels.ndim}'
+        'Expected logits.ndim==labels.ndim or logits.ndim==labels.ndim+1, '
+        f'got {logits.ndim} and {labels.ndim}'
       )
     elif labels.dtype in (jnp.int64, np.int32, np.int64):
       labels = jnp.astype(labels, jnp.int32)
     elif labels.dtype != jnp.int32:
       raise ValueError(f'Expected labels.dtype==jnp.int32, got {labels.dtype}')
 
+    if logits.ndim==labels.ndim:
```
Comment (cgarciae) [inline flax/nnx/training/metrics.py:273]: please add a comment here that this case is for binary classification

**Review by cgarciae — COMMENTED** [review]
Hey @mattbahr, thanks for doing this! Apart from the two other comments, I think `threshold` should be `None` by default and the user should explicitly opt-in to binary classification. This way we can have stronger shape checks for each case.

**Comment by mattbahr** [comment]
> Hey @mattbahr, thanks for doing this! Apart from the two other comments, I think `threshold` should be `None` by default and the user should explicitly opt-in to binary classification. This way we can have stronger shape checks for each case.

Good idea! Just made the requested changes. Let me know if there's anything else I can do to improve the implementation.

File: flax/nnx/training/metrics.py [hunk 3]
```diff
@@ -240,26 +240,64 @@ class Accuracy(Average):
...
+
+    >>> logits3 = jax.random.normal(jax.random.key(2), (5,))
+    >>> labels3 = jnp.array([0, 1, 0, 1, 1])
+    >>> accuracy = nnx.metrics.Accuracy(threshold=0.5)
+    >>> accuracy.update(logits=logits3, labels=labels3)
+    >>> accuracy.compute()
+    Array(0.8, dtype=float32)
   """
 
+  def __init__(self, threshold=None, *args, **kwargs):
```
Comment (cgarciae) [inline flax/nnx/training/metrics.py:252]: Add type tyhint:
```suggestion
  def __init__(self, threshold: float | None = None, *args, **kwargs):
```

**Review by cgarciae — APPROVED** [review]

File: flax/nnx/training/metrics.py (hunk 3 above)
Comment (mattbahr) [reply flax/nnx/training/metrics.py]: Type hint added
